[2024-05-21T18:32:43.872+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: fraud_bitcoin.train_decision_tree manual__2024-05-21T17:24:46.231076+00:00 [queued]>
[2024-05-21T18:32:43.908+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: fraud_bitcoin.train_decision_tree manual__2024-05-21T17:24:46.231076+00:00 [queued]>
[2024-05-21T18:32:43.910+0000] {taskinstance.py:1361} INFO - Starting attempt 3 of 3
[2024-05-21T18:32:43.969+0000] {taskinstance.py:1382} INFO - Executing <Task(PythonOperator): train_decision_tree> on 2024-05-21 17:24:46.231076+00:00
[2024-05-21T18:32:44.096+0000] {standard_task_runner.py:57} INFO - Started process 1411 to run task
[2024-05-21T18:32:44.116+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'fraud_bitcoin', 'train_decision_tree', 'manual__2024-05-21T17:24:46.231076+00:00', '--job-id', '370', '--raw', '--subdir', 'DAGS_FOLDER/fraud.py', '--cfg-path', '/tmp/tmpgrb8zf02']
[2024-05-21T18:32:44.147+0000] {standard_task_runner.py:85} INFO - Job 370: Subtask train_decision_tree
[2024-05-21T18:32:44.399+0000] {task_command.py:416} INFO - Running <TaskInstance: fraud_bitcoin.train_decision_tree manual__2024-05-21T17:24:46.231076+00:00 [running]> on host 66abfe055500
[2024-05-21T18:32:44.763+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='fraud_bitcoin' AIRFLOW_CTX_TASK_ID='train_decision_tree' AIRFLOW_CTX_EXECUTION_DATE='2024-05-21T17:24:46.231076+00:00' AIRFLOW_CTX_TRY_NUMBER='3' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-05-21T17:24:46.231076+00:00'
[2024-05-21T18:32:57.329+0000] {xcom.py:661} ERROR - Object of type DecisionTreeClassifier is not JSON serializable. If you are using pickle instead of JSON for XCom, then you need to enable pickle support for XCom in your *** config or make sure to decorate your object with attr.
[2024-05-21T18:32:57.343+0000] {taskinstance.py:1937} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/json.py", line 91, in default
    return serialize(o)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/serialization/serde.py", line 178, in serialize
    raise TypeError(f"cannot serialize object of type {cls}")
TypeError: cannot serialize object of type <class 'sklearn.tree._classes.DecisionTreeClassifier'>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 192, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 209, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/fraud.py", line 99, in train_decision_tree
    ti.xcom_push(key='dt_model', value=dt_model)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/session.py", line 79, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 2479, in xcom_push
    XCom.set(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/session.py", line 76, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/xcom.py", line 244, in set
    value = cls.serialize_value(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/xcom.py", line 659, in serialize_value
    return json.dumps(value, cls=XComEncoder).encode("UTF-8")
  File "/usr/local/lib/python3.8/json/__init__.py", line 234, in dumps
    return cls(
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/json.py", line 104, in encode
    return super().encode(o)
  File "/usr/local/lib/python3.8/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/local/lib/python3.8/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/json.py", line 93, in default
    return super().default(o)
  File "/usr/local/lib/python3.8/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type DecisionTreeClassifier is not JSON serializable
[2024-05-21T18:32:57.391+0000] {taskinstance.py:1400} INFO - Marking task as FAILED. dag_id=fraud_bitcoin, task_id=train_decision_tree, execution_date=20240521T172446, start_date=20240521T183243, end_date=20240521T183257
[2024-05-21T18:32:59.925+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 370 for task train_decision_tree (Object of type DecisionTreeClassifier is not JSON serializable; 1411)
[2024-05-21T18:33:00.183+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2024-05-21T18:33:00.277+0000] {taskinstance.py:2778} INFO - 0 downstream tasks scheduled from follow-on schedule check
